vars/githubscm.groovy

```diff
@@ -275,3 +275,27 @@ def findAndStageNotIgnoredFiles(String findNamePattern){
     git status
     """
 }
+
+def setDefaultBranch(String repo, String defaultBranch, String credentialId = 'kie-ci', String author) {
+    cleanHubAuth()
+    withCredentials([usernamePassword(credentialsId: credentialId, usernameVariable: 'GITHUB_USER', passwordVariable: 'GITHUB_TOKEN')]) {
+        // gh command from https://github.com/cli/cli/issues/929#issuecomment-629253585
+        def newDefaultBranch = sh(script: "../gh api -XPATCH 'repos/${author}/${repo}' -f default_branch=${defaultBranch} | jq '.default_branch'", returnStdout: true).trim()
```

Comment:
1/ `../gh` : does this utility assume the `gh` command is available in some particular directory? What if the working directory is different?
2/ where does the `jq` command come from? Is it available on the Jenkins machines OOTB?

Reply:
1. Good point let me add a condition on that
2. Jq is a json parser from github api, just to get the new default branch name. It is available after gh gets installed

Reply:
`gh` is installed there: https://github.com/kiegroup/kogito-examples/blob/d536c26b2163a871d69d70a39ba92f2a88022def/Jenkinsfile.promote#L282
maybe we could ask prod guys to have it installed and available directly on the node. wdyt ?

Reply:
Actually it is avalible at https://github.com/kiegroup/optaplanner/blob/bf3b9adb887a58b63cadd0fed96b67894d6effaf/Jenkinsfile.promote#L454 as well so in case u need that you need to install it in the jenkins job as I thought. It is not necessary to have it to build for nighties for example. So I added a check for that 57d5495

Reply:
Created a ticket to install gh on  nodes
https://issues.redhat.com/browse/BXMSPROD-1175